lib/usd/ui/layerEditor/componentSaveDialog.cpp

```diff
@@ -304,7 +311,38 @@ void ComponentSaveDialog::keyPressEvent(QKeyEvent* event)
     QDialog::keyPressEvent(event);
 }
 
-void ComponentSaveDialog::onSaveStage() { accept(); }
+void ComponentSaveDialog::onSaveStage()
+{
+    // Block overwriting of components. The target folder must be empty.
+    // Otherwise, log an error and abort. In the future we will want to
+    // support overwriting components. This is not trivial as we need
+    // to be able to preflight all the file write operations, and only if
+    // we are certain everything will succeed, overwrite everything. We also
+    // need to handle about-to-be-overwritten, but already in memory layers,
+    // locked layers, and a possibly polluted folder if the old component had
+    // assets that would no longer be used by the new version.
+
+    std::filesystem::path location = { _locationEdit->text().toStdString() };
+    location.append(_nameEdit->text().toStdString());
+
+    if (std::filesystem::exists(location) && !std::filesystem::is_empty(location)) {
```

Comment:
MayaUsd isn't C++17 because we have to support older versions of Maya. So you cannot use filesystem. Instead use gulrak. See other files for examples.